odoh.go

```diff
@@ -218,6 +198,55 @@ func (k ObliviousDoHConfigContents) CipherSuite() (hpke.CipherSuite, error) {
 	return hpke.AssembleCipherSuite(k.KemID, k.KdfID, k.AeadID)
 }
 
+type ObliviousDoHConfig struct {
+	Version  uint16
+	Contents ObliviousDoHConfigContents
+}
+
+func CreateObliviousDoHConfig(contents ObliviousDoHConfigContents) ObliviousDoHConfig {
+	return ObliviousDoHConfig{
+		Version:  ODOH_VERSION,
+		Contents: contents,
+	}
+}
+
+func (c ObliviousDoHConfig) Marshal() []byte {
+	marshalledConfig := c.Contents.Marshal()
+
+	buffer := make([]byte, 4)
+	binary.BigEndian.PutUint16(buffer[0:], uint16(c.Version))
+	binary.BigEndian.PutUint16(buffer[2:], uint16(len(marshalledConfig)))
+
+	configBytes := append(buffer, marshalledConfig...)
+	return configBytes
+}
+
+func UnmarshalObliviousDoHConfig(buffer []byte) (ObliviousDoHConfig, error) {
+	if len(buffer) < 4 {
+		return ObliviousDoHConfig{}, errors.New("Invalid ObliviousDoHConfig encoding")
+	}
+
+	version := binary.BigEndian.Uint16(buffer[0:])
+	if version != ODOH_VERSION {
```

Comment:
I disagree. Why would the library unmarshal to a type it literally does not support? The point of this check is to reject versions it can't possibly use.